maintenance/replaceTextEligible.php

```diff
@@ -0,0 +1,148 @@
+			$slotContentID = $dbr->newSelectQueryBuilder()
+				->select( [ 'slot_content_id' ] )
+				->from( 'slots' )
+				->where( [ 'slot_revision_id' => $page->page_latest ] )
+				->caller( __METHOD__ )->fetchRow();
+			$contentAddress = $dbr->newSelectQueryBuilder()
+				->select( [ 'content_address' ] )
+				->from( 'content' )
+				->where( [ 'content_id' => $slotContentID ] )
+				->caller( __METHOD__ )->fetchRow();
+			$oldID = substr( $contentAddress, 3 );
+			$textFlags = $dbr->newSelectQueryBuilder()
+				->select( [ 'old_flags' ] )
+				->from( 'text' )
+				->where( [ 'old_id' => $oldID ] )
+				->caller( __METHOD__ )->fetchRow();
+
+			if ( str_contains( $textFlags, 'gzip' ) ) {
+				// The latest revision of this page is compressed
+				$problematicPages[] = $page->page_name;
+			}
+		}
+
+		// Deleted pages
+		// These can be undeleted on-wiki, and if so, they may also cause issues with ReplaceText
+		$this->output( 'Processing deleted pages' );
+		foreach ( $deletedPageIDs as $deletedPageID ) {
+			// TODO: Use JOINs?
+			// Get the latest revision
+			$revID = $dbr->newSelectQueryBuilder()
+				->select( [ 'ar_rev_id' ] )
+				->from( 'archive' )
+				->where( [ 'ar_page_id' => $deletedPageID->ar_page_id ] )
+				->orderBy( 'ar_rev_id', SelectQueryBuilder::SORT_DESC )
+				->limit( 1 )
+				->caller( __METHOD__ )->fetchRow();
+			$slotContentID = $dbr->newSelectQueryBuilder()
+                                ->select( [ 'slot_content_id' ] )
+                                ->from( 'slots' )
+                                ->where( [ 'slot_revision_id' => $revID ] )
+                                ->caller( __METHOD__ )->fetchRow();
+                        $contentAddress = $dbr->newSelectQueryBuilder()
+                                ->select( [ 'content_address' ] )
+                                ->from( 'content' )
+                                ->where( [ 'content_id' => $slotContentID ] )
+                                ->caller( __METHOD__ )->fetchRow();
+                        $oldID = substr( $contentAddress, 3 );
+                        $textFlags = $dbr->newSelectQueryBuilder()
+                                ->select( [ 'old_flags' ] )
+                                ->from( 'text' )
+                                ->where( [ 'old_id' => $oldID ] )
+                                ->caller( __METHOD__ )->fetchRow();
+			if ( str_contains( $textFlags, 'gzip' ) ) {
+                                // The latest revision of this page is compressed
+				$deletedPageName = $dbr->newSelectQueryBuilder
+					->select( [ 'ar_page_name' ] )
+					->from( 'archive' )
+					->where( [ 'ar_page_id' => $deletedPageID->ar_page_id ] )
+					->limit( 1 )
+					->caller( __METHOD__ )->fetchRow();
+                                $problematicDeletedPages[] = $deletedPageName->ar_page_name;
+                        }
+		}
+		if ( count( $problematicPages ) > 0 || count( $problematicDeletedPages ) > 0 ) {
```

Comment:
This probably could be converted into an [early return](https://gomakethings.com/the-early-return-pattern-in-javascript/).